File: src/com/pailsom/leetcode/easy/Palindrome.java
package com.pailsom.leetcode.easy;

public class Palindrome {
    public static void main(String[] args) {
        System.out.println(isPalindrome(12344321));
    }

    private static boolean isPalindrome(int x) {
        if(x<0 || (x>0 && x %10==0)) return false;
        int reserveNum = 0;
        while(x>reserveNum){
            reserveNum = reserveNum*10 + x % 10;
            x= x /10;
        }
        return x==reserveNum || x == reserveNum/10;
    }

}

File: src/com/kotlin/leetcode/easy/Palindrome.kt
package com.kotlin.leetcode.easy

fun main() {
    println(isPalindrome(12321))
}

fun isPalindrome(num: Int): Boolean {
    var x = num
    if(x<0 || (x>0 && x%10==0)) return false
    var reserveNum:Int = 0
    while (x>reserveNum){
        reserveNum = reserveNum*10 + x % 10
        x /= 10
    }
    return x==reserveNum || (x==reserveNum/10)
}
